setup.py

```diff
@@ -19,6 +19,28 @@
     long_description = f.read()
 
 
+def is_list_of_dicts_with_keys(value, keys):
+    if isinstance(value, list):
+        return all(isinstance(item, dict) and all(key in item for key in keys) for item in value)
+    return False
+
+
+def format_installs_required(config):
+    dependencies = []
+    for k, v in config.items():
+        if k == "python":
+            continue
+        elif is_list_of_dicts_with_keys(v, ["version", "markers"]):
```

Comment:
What if it's a list of dicts where one entry doesn't have `version` or `markers`?
Should we just be checking the `package_version` for those keys and outputting them using the format using `f"{k}{version} ; {markers}"` if they are, or using `f"{k} ({v})"` if not?